tests/5.1/scope/test_scope_private_construct.c

```diff
@@ -0,0 +1,38 @@
+//--------------- test_scope_private_construct.c -----------------------------//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test checks that the scope private construct clause is properly working.
+// The test itself passes a test integer into the scope pragma and ensures that
+// all changes made to it are not kept outside of the scope region.
+//----------------------------------------------------------------------------//
+
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+#include <math.h>
+
+int test_scope(){
+	int errors;
+	int test_int = 1;
+	#pragma omp parallel firstprivate(test_int)
+	{
+		#pragma omp scope private(test_int)
+		{
+			test_int += 1;
+		}
+	}
+	OMPVV_TEST_AND_SET_VERBOSE(errors,test_int != 1);
+	OMPVV_INFOMSG_IF(test_int == 2, "test int was not private");
```

Comment:
I am not sure how 'test_int' can become '2':

* the parallel is 'firstprivate', thus, anything inside 'parallel' should not have any effect on the outside. → Use 'shared()' instead? — Currently, I always expect 1 on the outside.
* And with shared (and not working 'private' on scope: Shouldn't it be between 2 and (num-threads + 1), given that the code runs for num-threads (and the update is not done atomically).

Reply:
Your comments make a lot of sense, I went ahead and tried to fix this by first using shared() like you suggested as well as wrapping a task region instead a parallel one around the scope pragma in hopes on only having a single thread run the operations and negating the need to worry about any number of threads in the test check.